src/services/codefixes/inferFromUsage.ts

```diff
@@ -0,0 +1,655 @@
+                        const symbol = checker.createSymbol(SymbolFlags.Property, name);
+                        symbol.type = getTypeFromUsageContext(context, checker);
+                        members.set(name, symbol);
+                    });
+                }
+
+                if (usageContext.callContexts) {
+                    for (const callConext of usageContext.callContexts) {
+                        callSignatures.push(getSignatureFromCallContext(callConext, checker));
+                    }
+                }
+
+                if (usageContext.constructContexts) {
+                    for (const constructContext of usageContext.constructContexts) {
+                        constructSignatures.push(getSignatureFromCallContext(constructContext, checker));
+                    }
+                }
+
+                if (usageContext.numberIndexContext) {
+                    numberIndexInfo = checker.createIndexInfo(getTypeFromUsageContext(usageContext.numberIndexContext, checker), /*isReadonly*/ false);
+                }
+
+                if (usageContext.stringIndexContext) {
+                    stringIndexInfo = checker.createIndexInfo(getTypeFromUsageContext(usageContext.stringIndexContext, checker), /*isReadonly*/ false);
+                }
+
+                return checker.createAnonymousType(/*symbol*/ undefined, members, callSignatures, constructSignatures, stringIndexInfo, numberIndexInfo);
+            }
+            else {
+                return undefined;
+            }
+        }
+
+        function getParameterTypeFromCallContexts(parameterIndex: number, callContexts: CallContext[], isRestParameter: boolean, checker: TypeChecker) {
+            let types: Type[] = [];
+            if (callContexts) {
+                for (const callContext of callContexts) {
+                    if (callContext.argumentTypes.length > parameterIndex) {
+                        if (isRestParameter) {
+                            types = concatenate(types, map(callContext.argumentTypes.slice(parameterIndex), a => checker.getBaseTypeOfLiteralType(a)));
+                        }
+                        else {
+                            types.push(checker.getBaseTypeOfLiteralType(callContext.argumentTypes[parameterIndex]));
+                        }
+                    }
+                }
+            }
+
+            if (types.length) {
+                const type = checker.getWidenedType(checker.getUnionType(types, /*subtypeReduction*/ true));
+                return isRestParameter ? checker.createArrayType(type) : type;
+            }
+            return undefined;
+        }
+
+        function getSignatureFromCallContext(callContext: CallContext, checker: TypeChecker): Signature {
+            const parameters: Symbol[] = [];
+            for (let i = 0; i < callContext.argumentTypes.length; i++) {
+                const symbol = checker.createSymbol(SymbolFlags.FunctionScopedVariable, escapeLeadingUnderscores(`p${i}`));

```

Comment:
I would call this `arg${i}` to match our other names for unnamed parameters.

Reply:
done